src/web/components/ClickToView.tsx

```diff
@@ -89,74 +113,115 @@ const Headline = ({
-);
+	role: RoleType;
+}) => {
+	const textSize = roleTextSize(role);
+	return (
+		<div
+			className={css`
+				${textSize}
+				a {
+					${textSize}
+				}
+			`}
+		>
+			{children}
+		</div>
+	);
+};
+
+const AcceptButton = ({
+	role,
+	handleClick,
+}: {
+	role: RoleType;
+	handleClick: Function;
+}) => {
+	return (
+		<div>
+			<Button
+				priority="primary"
+				size={roleButtonSize(role)}
+				icon={<SvgCheckmark />}
+				iconSide="left"
+				onClick={() => handleClick()}
+			>
+				{roleButtonText(role)}
+			</Button>
+		</div>
+	);
+};
 
-export const ClickToView = ({ children, width, height, onAccept }: Props) => {
+export const ClickToView = ({
+	children,
+	role,
+	onAccept,
+	isTracking,
+	source,
+	sourceDomain,
+}: Props) => {
 	const [showOverlay, setShowOverlay] = useState<boolean>(true);
 
 	const handleClick = () => {
 		setShowOverlay(false);
-		onAccept();
+		if (onAccept) {
+			setTimeout(() => onAccept());
+		}
 	};
 
-	if (showOverlay) {
+	let headlineText;
+	let body;
+
+	const roleWithDefault: RoleType = role || 'inline';
+
+	if (isTracking && showOverlay) {
+		if (source) {
+			headlineText = `Allow ${source} content?`;
+			body = (
+				<div>
+					<p>
+						This article includes content provided by {source}. We
+						ask for your permission before anything is loaded, as
+						they may be using cookies and other technologies.
+					</p>
+					<p>
+						To view this content, click &apos;Allow and
+						continue&apos;.
+					</p>
+				</div>
+			);
+		} else {
+			headlineText = 'Allow content provided by a third party?';
+			body = (
+				<div>
+					<p>
+						This article includes content hosted on {sourceDomain}.
+						We ask for your permission before anything is loaded, as
+						the provider may be using cookies and other
+						technologies.
+					</p>
+					<p>
+						To view this content, click &apos;Allow and
+						continue&apos;.
+					</p>
+				</div>
+			);
+		}
+
 		return (
-			<Container width={width} height={height}>
-				<Outer>
-					<Inner>
-						<Top>
-							<Headline width={width}>
-								Allow content provided by a third party?
-							</Headline>
-							<Body width={width}>
-								<p>
-									This article includes content hosted on
-									other.com. We ask for your permission before
-									anything is loaded, as the provider may be
-									using cookies and other technologies.
-								</p>
-								<p>
-									To view this content, click &apos;Allow and
-									continue&apos;.
-								</p>
-							</Body>
-						</Top>
-						<Bottom>
-							<Button
-								priority="primary"
-								size={width > 300 ? 'small' : 'xsmall'}
-								icon={<SvgCheckmark />}
-								iconSide="left"
-								onClick={() => handleClick()}
-							>
-								Allow and continue
-							</Button>
-						</Bottom>
-					</Inner>
-					<Lines />
-				</Outer>
+			<Container>
+				<Headline role={roleWithDefault}>{headlineText}</Headline>
+				<Body role={roleWithDefault}>{body}</Body>
```

Comment:
My preference here would be to inline this html instead of having the `headlineText` and `body` variables. The reason  for this is to make future maintenance or refactors easier. In dotcom we strive to try and keep the code as simple as possible because we know we have a long path ahead of us involving lots of changes and refactors so the fewer abstractions and more inline code there is, the easier this will be. Obviously, this case is about as simple as abstractions get so it's true that the difference is very slight. The issue though is how the code might grow. Abstraction have a way of being built upon and this is the real thing we want to avoid.

Reply:
👍